pkg/registry/core/pod/storage/eviction.go

```diff
@@ -130,7 +130,15 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 
 	// Evicting a terminal pod should result in direct deletion of pod as it already caused disruption by the time we are evicting.
 	// There is no need to check for pdb.
-	if pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed {
+	if canIgnorePDB(pod) {
+		if shouldEnforceResourceVersion(pod) {
+			// Set deletionOptions.Preconditions.ResourceVersion to ensure we're not
+			// racing with another PDB-impacting process elsewhere.
+			if deletionOptions.Preconditions == nil {
+				deletionOptions.Preconditions = &metav1.Preconditions{}
+			}
+			deletionOptions.Preconditions.ResourceVersion = &pod.ResourceVersion
```

Comment:
by adding this precondition, you are indicating you want to get a conflict error if anything else has modified the pod since it was fetched on line 119. you would need to handle that conflict error here

Reply:
@liggitt what is your preferred way of handling this?  We could return 429 so clients will retry.

Reply:
@liggitt Do you want to return the 429 or leave as-is here?  If we do 429, drain client (and probably other clients) will retry here, which is probably what we want.

Reply:
I think we can just retry immediately, similar to what we do here https://github.com/kubernetes/kubernetes/blob/b94dca92a83447ad0384471c8a9ce017420714c1/pkg/registry/core/pod/storage/eviction.go#L164
We just need to make sure that we verify all the conditions again after getting the latest version of the pod from the api server.

Reply:
It's not entirely clear what that code is doing, the comments look stale.  I'll dig in and see what it's actually doing, hopefully there are tests.

Reply:
Okay, I sorted out what that code is doing, it does indeed return a 429 at appropriate times, but it doesn't have good test coverage.

I went ahead and did something similar for the ignorePDB delete call.